proxy/coco/src/peer.rs

```diff
@@ -715,6 +752,96 @@ mod test {
         Ok(())
     }
 
+    /// Verify that asking the network for an unkown urn returns no providers.
+    #[tokio::test]
+    async fn get_urn_providers_none() -> Result<(), Error> {
+        let tmp_dir = tempfile::tempdir().expect("failed to create temdir");
+        let key = SecretKey::new();
+        let config = config::default(key, tmp_dir.path())?;
+        let api = Api::new(config).await?;
+        let unkown_urn = uri::RadUrn {
+            id: Hash::hash(b"project0"),
+            proto: uri::Protocol::Git,
+            path: "user/imperative-language"
+                .parse::<uri::Path>()
+                .map_err(Error::from)?,
+        };
+
+        let res = timeout(
+            Duration::from_secs(5),
+            api.providers(unkown_urn).await.await.next(),
+        )
+        .await;
+
+        assert!(
+            res.is_err(),
+            "Should have timed out without obtaining any provider",
+        );
+
+        Ok(())
+    }
+
+    /// Verify that asking the network for a URN owned by a seed peer returns said peer.
```

Comment:
```suggestion
```

Reply:
Could you elaborate on why? I find such small doc comments to be super useful if you haven't looked into a piece in a while, while not having it only saves some bytes. 

Same for the suggested deletions above. The test gets pretty big, adding those small annotations is pretty helpful in my experience.

Reply:
As mentioned in other places, if the code doesn't communicate what these small comments allude to, it needs to change. Here the test function should give indication enough that a comment is not necessary.

Reply:
I used to be an extreme believer in that principle until my pursue for that principle got in the way of other developers and their experience.

I'd welcome suggestions to make the code so that readers don't benefit from these small annotations while making sense from a trade-off pov.

Reply:
> Here the test function should give indication enough that a comment is not necessary.

In all the best spirit, I see a substantial inconsistency between this point :point_up: and requiring _every_ `struct` or `function` to be documented, where said documentation is absolutely redundant and adds no useful information nor value. Just a few examples are [this](https://github.com/radicle-dev/radicle-upstream/blob/3377508a3d01f8638a921d23244256b6c17580c9/proxy/api/src/avatar.rs#L90) and [this](https://github.com/radicle-dev/radicle-upstream/blob/3377508a3d01f8638a921d23244256b6c17580c9/proxy/api/src/avatar.rs#L203) (nothing against `avatar.rs` :laughing:)

Reply:
When I read the code I didn't benefit from the comments and therefore suggested the changes. The code as is was clear enough to follow. It seems more like a liability in this instance. Then again I'm familiar with the code-base and most of it's darker corners, so take it with a grain of salt. Wouldn't be fuzzed if you don't accept the suggestions, :dog:

Reply:
> Wouldn't be fuzzed if you don't accept the suggestions, :dog: 

Ok :) in that case, I appreciate the sensitivity in accepting other experiences here.